iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string_internal.hpp

```diff
@@ -73,6 +77,7 @@ struct GetSize<string<N>>
 };
 
 template <uint64_t N>
+// NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays) cxx::string wraps char array
```

Comment:
I would state, used to acquire size of c array safely, `strnlen` only accesses `N` elements which is the maximum capacity of the array. Hereby is `N` a compile time constant.